provider/pkg/await/await.go

```diff
@@ -531,18 +599,18 @@ func fixCSAFieldManagers(c *UpdateConfig, live *unstructured.Unstructured, clien
 			},
 		}, c.FieldManager)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		patch, err := json.Marshal(patches)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		_, err = client.Patch(c.Context, live.GetName(), types.JSONPatchType, patch, metav1.PatchOptions{})
```

Comment:
Shouldn't we update the `live` variable here rather than discarding the new live object from `Patch`? Imagine that the managers were indeed migrated; you'd want to know that in the subsequent `handleSSAIgnoreFields` step.

If not, then I don't understand why the signature of `fixCSAFieldManagers` was changed to suggest that a new live object is returned.

Reply:
Good catch! This happened after I rebased on top of merged changes to mainline branch. I've refactored the code slightly in this function to make returning a new unstructured.Unstructured instead of re-using the `live` var name to prevent an error like this from happening again.